tool-metagen/src/DA/Daml/GenDaml.hs

```diff
@@ -32,12 +32,22 @@ ppImports = vcat . List.intersperse (text "") . map ppImport
     ppImport (Qualified modu prefix) =
         text "import qualified" <+> text modu <+> text "as" <+> text prefix
 
-ppDecls :: Show a => [Decl a] -> Doc
-ppDecls = vcat . List.intersperse (text "") . map ppDecl
+ppDecls :: Show a => Bool -> [Decl a] -> Doc
+ppDecls useGhcPrimitives = vcat . List.intersperse (text "") . map (ppDecl useGhcPrimitives)
 
-ppDecl :: Show a => Decl a -> Doc
+derivingDecl :: Bool -> String -> Doc
```

Comment:
I'm not sure what this is all about. Can you explain?

Regardless, I would clean this up a bit. For clarity I would remove this function, and inline it where it's used (line 63); it's confusing for instance that in line 50 you also have a `deriving (Eq, Ord Show)`. Also you'll be able to get rid of all those `_useGhcPrimitives` names at the signature level, which are unused in all but one declaration.

But I suspect I'm not understanding the purpose of this so the above may be invalid.

Reply:
I am passing a flag down through these pretty print instructions to control whether we add a `deriving (Eq, Ord, Show)` line to our data types or to use custom instances based off of GHC primitives.

Agreed, the function makes more sense closer to the usage.

Reply:
> to use custom instances based off of GHC primitives.

That's the bit I don't understand. Why do you need to do that?

Reply:
To avoid recursion limits. For another project we were not able to upload the Xml contents because `Show` would break a (gRPC or Java, i think) recursion limit.